unstructured/embed/mixedbreadai.py

```diff
@@ -0,0 +1,214 @@
+import os
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, List, Optional
+
+import numpy as np
+
+from unstructured.documents.elements import Element
+from unstructured.embed.interfaces import BaseEmbeddingEncoder, EmbeddingConfig
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+
+USER_AGENT = "@mixedbread-ai/unstructured"
+MAX_BATCH_SIZE = 256
+
+if TYPE_CHECKING:
+    from mixedbread_ai.client import MixedbreadAI
+    from mixedbread_ai.core import RequestOptions
+    from mixedbread_ai.types import EncodingFormat, TruncationStrategy
+
+
+@dataclass
+class MixedbreadAIEmbeddingConfig(EmbeddingConfig):
+    """
+    Configuration class for Mixedbread AI Embedding Encoder.
+
+    Attributes:
+        api_key (str): API key for accessing Mixedbread AI.
+        base_url (Optional[str]): Base URL for the API.
+        timeout (Optional[int]): Timeout for API requests in seconds.
+        max_retries (Optional[int]): Maximum number of retries for API requests.
+        batch_size (Optional[int]): Batch size of API requests.
+        model_name (str): Name of the model to use for embeddings.
+        normalized (bool): Whether to normalize the embeddings.
+        encoding_format (Optional[EncodingFormat]): Format of the encoding.
+        truncation_strategy (Optional[TruncationStrategy]): Strategy for truncating text.
+        dimensions (Optional[int]): Number of dimensions for the embeddings.
+        prompt (Optional[str]): Optional prompt for embedding generation.
+    """
+
+    api_key: str = field(
+        default_factory=lambda: os.environ.get("MXBAI_API_KEY", None),
+    )
+    base_url: Optional[str] = field(default=None)
+    timeout: Optional[int] = field(default=60)
+    max_retries: Optional[int] = field(default=3)
+    batch_size: Optional[int] = field(default=128)
+
+    model_name: str = field(
+        default="mixedbread-ai/mxbai-embed-large-v1",
+    )
+    normalized: bool = field(default=True)
+    encoding_format: Optional["EncodingFormat"] = field(default="float")
+    truncation_strategy: Optional["TruncationStrategy"] = field(default="start")
+    dimensions: Optional[int] = field(default=None)
```

Comment:
I've never seen an embedder that doesn't want the embedding_dimension. (thought I've only used the embedders in unstructured...) Is there a default embedding dimension?

Reply:
Yes, the dimensions is default to the models' maximum if you don't pass in. The reason why we have this argument is some of our embedding models support Matryoshka and they allow a smaller number of dimensions for the embeddings, while retain most of the performance.

Reply:
Aah. Sorry. We set the embedding dimension by setting the model. I see you do have a default model.